src/Controller/CustomerController.php

```diff
@@ -32,21 +36,30 @@ public function __construct(CustomerRepository $customerRepository)
 
      public function addCustomer(Request $request): JsonResponse
      {
-         $data = json_decode($request->getContent(), true);
+        
+            $data = json_decode($request->getContent(), true);
+        
+            try {
+                $firstName = $data["firstName"];
+                $lastName = $data["lastName"];
+                $email = $data["email"];
+                $phoneNumber = $data["phoneNumber"];
+            } catch (ErrorException $e) {
+                return new JsonResponse(["message" => "Enter value for all field"], Response::HTTP_BAD_REQUEST);
+            }
+
+            if (empty($firstName) || empty($lastName) || empty($email) || empty($phoneNumber)){
+                throw new NotFoundHttpException("Expecting mandatory parameters:");
+            }
```

Comment:
You can add these validation for `$data` itself. Instead of assigning to variable and then checking.
Something like this:
```suggestion
            if (empty($data["firstName"]) 
                || empty($data["lastName"]) 
                || empty($data["email"]) 
                || empty($data["phoneNumber"])) {
                throw new NotFoundHttpException("Expecting mandatory parameters:");
            }
            try {
                $firstName = $data["firstName"];
                $lastName = $data["lastName"];
                $email = $data["email"];
                $phoneNumber = $data["phoneNumber"];
            } catch (ErrorException $e) {
                return new JsonResponse(["message" => "Enter value for all field"], Response::HTTP_BAD_REQUEST);
            }
```